core/src/main/java/pt/webdetails/cpf/utils/AbstractCorsUtil.java

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Copyright 2002 - 2017 Webdetails, a Pentaho company. All rights reserved.
+ *
+ * This software was developed by Webdetails and is provided under the terms
+ * of the Mozilla Public License, Version 2.0, or any later version. You may not use
+ * this file except in compliance with the license. If you need a copy of the license,
+ * please go to http://mozilla.org/MPL/2.0/. The Initial Developer is Webdetails.
+ *
+ * Software distributed under the Mozilla Public License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. Please refer to
+ * the license for the specific language governing your rights and limitations.
+ */
+
+package pt.webdetails.cpf.utils;
+
+import org.apache.commons.lang.StringUtils;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Is used to implement cross-domain calls for an embedded scenario
+ */
+public abstract class AbstractCorsUtil {
+
+  private static final String TRUE = "true";
+  private static final String ORIGIN = "ORIGIN";
+  private static final String ACCESS_CONTROL_ALLOW_ORIGIN = "Access-Control-Allow-Origin";
+  private static final String ACCESS_CONTROL_ALLOW_CREDENTIALS = "Access-Control-Allow-Credentials";
+  private static final String COMMA = ",";
+
+  /**
+   * @return "true" if cross-domain calls are allowed
+   */
+  protected abstract String getAllowCrossDomainResources();
+
+  /**
+   * @return comma separated list of allowed domains
+   */
+  protected abstract String getDomainWhitelist();
```

Comment:
Each of the plugins will load it from settings.xml which returns String, so I will agre with you if we will leave the parsing logic in an abstarct class. Each plugin will have an ability to use it but will not be forced.